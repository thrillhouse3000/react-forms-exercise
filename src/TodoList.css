.TodoList-list {
    list-style: none;
}