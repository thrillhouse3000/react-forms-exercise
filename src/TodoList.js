import {useState} from "react"
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import {v4 as uuid} from "uuid"
import './TodoList.css'

const TodoList = () => {
    const INITIAL_STATE = [];
    const [todos, setTodos] = useState(INITIAL_STATE)

    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, {...newTodo, id: uuid()}])
    }

    const removeTodo = (id) => {
        setTodos(todos.filter(t => t.id !== id))
    }

    return (
        <div className='TodoList'>
            <h2>Todo List</h2>
            <NewTodoForm addTodo={addTodo} />
            <div className='TodoList-list'>
                {todos.map(t => <Todo id={t.id} key={t.id} todo={t.todo} removeTodo={removeTodo} />)}
            </div>
        </div>
    )
};

export default TodoList;