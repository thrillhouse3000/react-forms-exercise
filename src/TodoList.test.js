import { render, screen, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it('renders without crashing', () => {
    render(<TodoList />)
});

it('matches snapshot', () => {
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot()
});

it('adds a todo', () => {
    render(<TodoList />)
    const input = screen.getByLabelText('Todo')
    const addBtn = screen.queryByText('Add Todo')
    let todoBtn = screen.queryByText('X')
    expect(todoBtn).not.toBeInTheDocument()

    fireEvent.change(input, {target: {value: 'Test'}})
    fireEvent.click(addBtn)

    todoBtn = screen.queryByText('X')
    expect(todoBtn).toBeInTheDocument()
})

it('deletes a todo', () => {
    render(<TodoList />)
    const input = screen.getByLabelText('Todo')
    const addBtn = screen.queryByText('Add Todo')
    let todoBtn = screen.queryByText('X')
    expect(todoBtn).not.toBeInTheDocument()

    fireEvent.change(input, {target: {value: 'Test'}})
    fireEvent.click(addBtn)

    todoBtn = screen.queryByText('X')
    expect(todoBtn).toBeInTheDocument()

    fireEvent.click(todoBtn)
    expect(todoBtn).not.toBeInTheDocument()
})