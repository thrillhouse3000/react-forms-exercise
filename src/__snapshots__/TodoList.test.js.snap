// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h2>
      Todo List
    </h2>
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Enter a Todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div
      class="TodoList-list"
    />
  </div>
</DocumentFragment>
`;
